kore/src/Kore/JsonRpc.hs

```diff
@@ -561,6 +558,9 @@ respond serverState moduleName runSMT =
                                         ServerState
                                             { serializedModules =
                                                 Map.insert (coerce name) foundSerModule serializedModules
+                                            , receivedModules = case Map.lookup (coerce moduleHash) receivedModules of
+                                                Just recMod -> Map.insert (coerce name) recMod receivedModules
+                                                Nothing -> receivedModules
```

Comment:
I wonder if this `Nothing` case is actually possible... :thinking: if we have already found the module hash in the indexed modules we must have received it via `add-module` - unless someone loads a maliciously-named module on startup, another corner case :smiley_cat: )

Reply:
yeah :/